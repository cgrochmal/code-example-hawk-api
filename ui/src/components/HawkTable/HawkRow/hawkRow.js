import React from 'react'
import PropTypes from 'prop-types'

import './hawkRow.scss'

/**
 * Helper Component for <HawkTable> - renders an individual row
 */

HawkRow.propTypes = {
  hawk: PropTypes.object.isRequired,
  handleViewHawk: PropTypes.func,
  selectedHawk: PropTypes.object
}

export default function HawkRow(props) {
  const {hawk, handleViewHawk, selectedHawk} = props
  const isSelected = selectedHawk && selectedHawk.id === hawk.id
  const buttonText = isSelected ? 'Close' : 'View'
  return (
    <div className='hawk-row'>
      <span className='hawk-row__cell hawk-row__cell--name'>
        {hawk.name}
      </span>
      <span className='hawk-row__cell hawk-row__cell--size'>
        {hawk.size}
      </span>
      <span className='hawk-row__cell hawk-row__cell--gender'>
        {hawk.gender}
      </span>
      <span className='hawk-row__cell hawk-row__cell--view'>
        <button className='view-hawk-button' onClick={() => handleViewHawk(hawk)}>{buttonText}</button>
      </span>
    </div>
  )
}