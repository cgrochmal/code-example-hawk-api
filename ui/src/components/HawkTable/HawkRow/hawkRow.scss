.hawk-row {
  .view-hawk-button {
    border-radius: 5px;
  }
  &__cell {
  }
}