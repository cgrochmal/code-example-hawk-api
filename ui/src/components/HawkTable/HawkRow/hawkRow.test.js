import React from 'react'
import renderer from 'react-test-renderer'
import HawkRow from './hawkRow'
import TestHelper from '../../../test/testHelper'

it('matches the HawkRow snapshot', () => {
  const mockHawks = TestHelper.getMockHawks(1)
  const tree = renderer
    .create(<HawkRow hawk={mockHawks[0]}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})