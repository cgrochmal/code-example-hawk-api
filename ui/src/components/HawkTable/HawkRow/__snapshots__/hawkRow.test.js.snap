// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the HawkRow snapshot 1`] = `
<div
  className="hawk-row"
>
  <span
    className="hawk-row__cell hawk-row__cell--name"
  >
    test0
  </span>
  <span
    className="hawk-row__cell hawk-row__cell--size"
  >
    SMALL
  </span>
  <span
    className="hawk-row__cell hawk-row__cell--gender"
  >
    FEMALE
  </span>
  <span
    className="hawk-row__cell hawk-row__cell--view"
  >
    <button
      className="view-hawk-button"
      onClick={[Function]}
    >
      View
    </button>
  </span>
</div>
`;
